src/util/algorithms.cpp

```diff
@@ -38,4 +38,37 @@ bool goto_functions_algorithm::runOnFunction(
   std::pair<const dstring, goto_functiont> &)
 {
   return true;
-}
\ No newline at end of file
+}
+
+bool ssa_algorithm::run()
+{
+  // Helper definition
+  typedef goto_trace_stept::typet ssa_type;
+
+  // Iterates over all steps calling the specific function based on the step.type
+  for(auto &step : steps)
+    if(should_parse_step(step))
+      switch(step.type)
+      {
+      case ssa_type::ASSIGNMENT:
+        run_on_assignment(step);
+        break;
+      case ssa_type::ASSUME:
+        run_on_assume(step);
+        break;
+      case ssa_type::ASSERT:
+        run_on_assert(step);
+        break;
+      case ssa_type::OUTPUT:
+        run_on_output(step);
+        break;
+      case ssa_type::RENUMBER:
+        run_on_renumber(step);
+        break;
+      case ssa_type::SKIP:
+        run_on_skip(step);
+        break;
```

Comment:
missing default?

Reply:
I don't think `default:` is missing: what would you do in there? The way it is now, we'll get a compiler warning when the `ssa_type` enum changes. IMHO that's better than either aborting (making the compiled code slower) or just `break`, which only hides the warning.